test/viewmodels/comment_view_model_integration_test.dart

```diff
@@ -0,0 +1,149 @@
+      postId = post.id;
+
+      commentViewModelProvider = commentListProvider(postId);
+    });
+
+    group("display of comments", () {
+      test("No comments are returned when the database is empty", () async {
+        final comments = await container.read(commentViewModelProvider.future);
+
+        expect(comments, isEmpty);
+      });
+
+      test("Multiple comments are returned when the database has comments",
+          () async {
+        const numberOwners = 3;
+        const numberComments = 19;
+
+        // Create and add the comments owner to the database
+        final owners =
+            FirestoreUserGenerator.generateUserFirestore(numberOwners);
+        for (final owner in owners) {
+          await userRepository.setUser(owner.uid, owner.data);
+        }
+
+        // Create and add the comments to the database
+        final commentsData = commentDataGenerator
+            .generateCommentData(numberComments)
+            .mapIndexed(
+              (index, commentData) => CommentData(
+                ownerId: owners[index % numberOwners].uid,
+                publicationTime: commentData.publicationTime,
+                voteScore: commentData.voteScore,
+                content: commentData.content,
+              ),
+            )
+            .toList();
+
+        final expectedComments = <CommentPost>[];
+
+        for (var i = 0; i < numberComments; i++) {
+          final commentData = commentsData[i];
+          final owner = owners[i % numberOwners];
+
+          await commentRepository.addComment(
+            postId,
+            commentData,
+          );
+
+          final comment = CommentPost.from(commentData, owner.data);
+
+          expectedComments.add(comment);
+        }
+
+        final actualComments =
+            await container.read(commentViewModelProvider.future);
+
+        // Check that all the comments are present
+        expect(actualComments, unorderedEquals(expectedComments));
+
+        final expectedSortedComments = expectedComments.sorted(
+          (commentA, commentB) =>
+              -commentA.publicationDate.compareTo(commentB.publicationDate),
+        );
+
+        // Check that the comments are sorted from the newest to the oldest
+        expect(actualComments, expectedSortedComments);
+      });
+    });
+
+    group("refresh logic", () {
+      test("refresh shows newly added comment", () async {
+        // Before the refresh, there should be no comments
+        final comments = await container.read(commentViewModelProvider.future);
+
+        expect(comments, isEmpty);
+
+        // Then a comment is added
+        final owner = FirestoreUserGenerator.generateUserFirestore(1).first;
+        await userRepository.setUser(owner.uid, owner.data);
+
+        final commentData =
+            commentDataGenerator.createMockCommentData(ownerId: owner.uid);
+        await commentRepository.addComment(
+          postId,
+          commentData,
+        );
+
```

Comment:
I think verifying here that there is no comment is also interesting. I'm not entierly sure that the verification on lines 118-120 is really that useful, but this is your implementation choice to move it here or copy it here.

Reply:
Yes this is a very good point :+1: 
I have moved the check accordingly.
Note that I still need to call `await container.read(commentViewModelProvider.future);` a first time at the beginning of the test to force a first call to the `build()` method of the viewmodel that will expose the empty state.

Then I check that that this state was not updated by adding a comment to the repository.

Implemented in a778d3a188395f4a689740ae75ab74fe4ec50ee0.